appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaFormEditor.java

```diff
@@ -694,7 +695,16 @@ private MockComponent createMockComponent(JSONObject propertiesObject, MockConta
     // Set component properties
     for (String name : properties.keySet()) {
       if (name.charAt(0) != '$') { // Ignore special properties (name, type and nested components)
-        mockComponent.changeProperty(name, properties.get(name).asString().getString());
+        JSONValue j = properties.get(name);
+        if (j instanceof JSONString) {
+          mockComponent.changeProperty(name, j.asString().getString());
+        } else if (j instanceof JSONArray){
+          for (JSONValue nestedComponent : properties.get(j).asArray().getElements()) {
+            createMockComponent(nestedComponent.asObject(), (MockContainer) mockComponent, substitution);
+          }
+        }else {
+            mockComponent.changeProperty(name, j.toJson());
+        }
```

Comment:
Hmm. I am not sure about this logic. All properties in App Inventor are stored directly as strings in the Screen JSON. When would this be applied?